src/utilities/selection/Selection.ts

```diff
@@ -15,16 +22,25 @@ export class Selection implements ISelection {
   private _anchoredIndex: number;
   private _onSelectionChanged: () => void;
   private _hasChanged: boolean;
-
-  constructor(onSelectionChanged?: () => void, getKey?: (item: IObjectWithKey, index?: number) => string) {
+  private _dontSelectIndices: { [index: string]: boolean };
```

Comment:
The name prefix _dont make it weird to understand intent here.

How about _unselectableIndices _unselectableCount